lib/rules/indent.js

```diff
@@ -106,43 +108,51 @@ module.exports = function(context) {
          * @private
          */
         function getFixerFunction() {
-            var rangeToFix = [];
+            var replacement = "" + new Array(needed + 1).join(indentChar),  // replace with repeat in future
+                endOfPreviousToken = -1,
+                endOfLeadingComments = -1,
+                previousNewlineIndex = -1,
+                rangeToFix,
+                startOfLine,
+                referenceToken,
+                previousToken,
+                leadingComments;
+
+            if (isLastNodeCheck === true) {
+                referenceToken = sourceCode.getLastToken(node);
+            } else {
+                referenceToken = sourceCode.getFirstToken(node);
+            }
 
-            if (needed > gotten) {
-                var spaces = "" + new Array(needed - gotten + 1).join(indentChar);  // replace with repeat in future
+            // Three candidates for "start of line" for indent purposes.
 
-                if (isLastNodeCheck === true) {
-                    rangeToFix = [
-                        node.range[1] - 1,
-                        node.range[1] - 1
-                    ];
-                } else {
-                    rangeToFix = [
-                        node.range[0],
-                        node.range[0]
-                    ];
-                }
+            // 1. End of leading comment.
+            leadingComments = sourceCode.getComments(node).leading;
```

Comment:
Yeah, I mean....you can probably leave it and I can tackle it when i work on indenting of comments.